### PR TITLE
Add dynamic value to betting

### PR DESCRIPTION
**Pull request by chadellison**
There are a couple of things I would love feedback for on this PR, especially from anyone who has experimented with sockets and or rails 5.

I have established a socket connection between the server and multiple clients who subscribe to the room_channel.

I have event listeners on these buttons on the browser-side in this file: app/assets/javascripts/channels/room.coffee

On the server-side in "app/channels/room_channel.rb", I am doing this:

def speak(data)
   # ActionCable.server.broadcast "room_channel", message: data["message"]
   #have access to current user here
   Message.create! content: "#{current_user.username}: #{data["message"]}"
end

Here I have access to whichever user hits a button through current_user. 
I am planning on manipulating the state of the user from here, the room_channel.rb. My question is whether or not this is good practice. Is there another way I should be moving from the DOM to the database. 

Any insight here would be useful. Thanks for taking the time to look over my code.

### CODE REVIEW

**Comment by theonlyrao** [comment]
It's really cool you got this working! There's definitely a lot of moving pieces. And I don't know anything about best practices.

However, from what I understand of this implementation, it looks like a user's action to click a button goes straight to the db through the `speak` method in `room_channel.rb` ([here](https://github.com/chadellison/texas_holdem/blob/master/app/channels/room_channel.rb#L11-L14)).

I think this makes a lot of sense as an implementation right now when you are trying to figure out how all this works. I think I would be concerned about this as creating a lot of technical debt for the future. What comes to mind first is the fact that sometimes a user will click the button but won't actually be allowed to take the action - for example, trying to fold out of turn or perhaps betting less than what's already been raised.

If the broadcasted message is called by the clicking of the button directly, then it seems like you'd have to have all kinds of conditionals or something in place to make sure that messages aren't broadcasted when the underlying action is impermissible.

So it seems to me that now that you've got the broadcasting figured out, you might want to build out a slice of the actual gameplay and see how all the validating of user actions and db interactions are going to work together. Then once you've got all the objects sorted out, you could figure out which one of those should tell the room that it's time to speak. 

But again, nice job getting the messaging to work!

**Comment by chadellison** [comment]
Hey thanks so much Ashwin. That is very helpful and I definitely see your point about the technical debt. I want to start connecting some game logic soon to see what I've gotten myself into.

File: app/channels/room_channel.rb [hunk 1]
```diff
@@ -11,6 +11,15 @@ def unsubscribed
   def speak(data)
     # ActionCable.server.broadcast "room_channel", message: data["message"]
     #have access to current user here
-    Message.create! content: "#{current_user.username}: #{data["message"]}"
+    client_action = data["message"]
+    binding.pry
```
Comment (mattr-) [inline app/channels/room_channel.rb:15]: be wary of committing `binding.pry` lines to the main repo. other people on your team will not appreciate having to figure out why the server's taking so long to return something :smiley:

**Comment by chadellison** [comment]
Ahh snap! Thanks Mark!

**Pull request merged**